### PR TITLE
enhance #169: integrate linters into CI

### PR DESCRIPTION
**Pull request by cncal**
# Describe what this PR does / why we need it

Integrate golang linters (**govet**, **staticcheck**, **ineffassign** and **misspell**) into CI. Those linters is commonly used I think and you can find more available ones [here](https://golangci-lint.run/usage/linters/).

### Does this pull request fix one issue?

Enhance #169

### Describe how you did it

Update `.travis.yml`.

### CODE REVIEW

**Comment by louyuting** [comment]
The CI seems to stop run.

**Comment by cncal** [comment]
It seems that travis has completed your latest PR building and you can check it out [here](https://travis-ci.org/github/alibaba/sentinel-golang). I think maintainers should check the project settings to find out why it doesn't show in PR detail.

File: .travis.yml [hunk 1]
```diff
@@ -1,21 +1,29 @@
...
-before_install: go get golang.org/x/tools/cmd/goimports
+before_install:
+  - go get golang.org/x/tools/cmd/goimports
+  - curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.27.0
 
 script:
   - diff -u <(echo -n) <(gofmt -d -s .)
   - diff -u <(echo -n) <(goimports -d .)
   - go test -v -race ./... -coverprofile=coverage.txt -covermode=atomic
+  - golangci-lint run -v --disable-all
```
Comment (sczyh30) [inline .travis.yml:21]: Maybe we could put the linter before running `go test`?

**Comment by sczyh30** [comment]
I've re-triggered the CI and it now appears in the checklist.

File: .travis.yml (hunk 1 above)
Comment (cncal) [reply .travis.yml:21]: All right. I'll modify it

**Comment by louyuting** [comment]
Pending merge by dubbo-go issue

File: adapter/dubbo/util.go [hunk 2]
```diff
@@ -9,15 +9,17 @@ import (
 	"github.com/apache/dubbo-go/protocol"
 )
 
+type contextKey string
```
Comment (louyuting) [inline adapter/dubbo/util.go:12]: It's confusing, but staticcheck lint here

File: core/flow/slot.go [hunk 3]
```diff
@@ -52,6 +52,7 @@ func canPassCheck(tc *TrafficShapingController, node base.StatNode, acquireCount
 func canPassCheckWithFlag(tc *TrafficShapingController, node base.StatNode, acquireCount uint32, flag int32) *base.TokenResult {
 	if tc.rule.ClusterMode {
 		// TODO: support cluster mode
+		logging.Warnf("unsupported cluster mode")
```
Comment (louyuting) [inline core/flow/slot.go:55]: staticcheck lint here

File: core/flow/tc_warm_up.go [hunk 4]
```diff
@@ -16,9 +16,9 @@ type WarmUpTrafficShapingCalculator struct {
 	warmUpPeriodInSec uint32
 	coldFactor        uint32
 	warningToken      uint64
-	maxToken          uint64
+	maxToken          int64
```
Comment (louyuting) [inline core/flow/tc_warm_up.go:19]: Need to discuss here.
This changes just is a workround

File: core/flow/tc_warm_up.go [hunk 5]
```diff
@@ -71,28 +71,29 @@ func (d *WarmUpTrafficShapingCalculator) syncToken(passQps float64) {
 		return
 	}
 
-	oldValue := atomic.LoadUint64(d.storedTokens)
+	oldValue := atomic.LoadInt64(d.storedTokens)
 	newValue := d.coolDownTokens(currentTime, passQps)
 
-	if atomic.CompareAndSwapUint64(d.storedTokens, oldValue, newValue) {
-		if currentValue := atomic.AddUint64(d.storedTokens, uint64(0-passQps)); currentValue < 0 {
```
Comment (louyuting) [inline core/flow/tc_warm_up.go:78]: currentValue is uint64(meaning >=0)
Here is a bug?

File: core/hotspot/rule_manager_test.go [hunk 6]
```diff
@@ -34,7 +34,9 @@ func Test_tcGenFuncMap(t *testing.T) {
 		tc := generator(r1, nil)
 		assert.True(t, tc.BoundMetric() != nil && tc.BoundRule() == r1 && tc.BoundParamIndex() == 0)
 		rejectTC := tc.(*rejectTrafficShapingController)
-		assert.True(t, rejectTC != nil)
+		if rejectTC == nil {
```
Comment (louyuting) [inline core/hotspot/rule_manager_test.go:37]: confuding, but govet lint here

File: adapter/dubbo/util.go (hunk 2 above)
Comment (louyuting) [reply adapter/dubbo/util.go:12]: https://blog.golang.org/context#TOC_3.2.
refer to

**Review by louyuting — APPROVED** [review]
Thanks for your contribution

**Pull request merged**